main/integration-tests/__test__/advanced-tests/study-permissions/verify-linux-study-permissions.test.js

```diff
@@ -0,0 +1,204 @@
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+const { sleep } = require('@aws-ee/base-services/lib/helpers/utils');
+const { NodeSSH } = require('node-ssh');
+const { mountStudies, readWrite } = require('../../../support/complex/run-shell-command');
+const { runSetup } = require('../../../support/setup');
+
+describe('EC2 Linux scenarios', () => {
+  let setup;
+  let ssh;
+  async function testSetup() {
+    const adminSession = await setup.createAdminSession();
+    const admin2Session = await setup.createAdminSession();
+    const keyPair = await admin2Session.resources.keyPairs.create();
+    return { adminSession, admin2Session, keyPair };
+  }
+
+  beforeAll(async () => {
+    setup = await runSetup();
+    ssh = new NodeSSH();
+  });
+  afterAll(async () => {
+    await setup.cleanup();
+  });
+
+  describe('Updates to mounted study permissions', () => {
+    it('should propagate for Org Study', async () => {
+      const { adminSession, admin2Session, keyPair } = await testSetup();
+      const studyId = setup.gen.string({ prefix: `create-org-study-test` });
+      await adminSession.resources.studies.create({ id: studyId, name: studyId, category: 'Organization' });
+      await adminSession.resources.studies
+        .study(studyId)
+        .propagatePermission(admin2Session, ['admin', 'readwrite'], []);
+
+      const workspaceName = setup.gen.string({ prefix: 'workspace-sc-test' });
+
+      const env = await admin2Session.resources.workspaceServiceCatalogs.create({
+        name: workspaceName,
+        envTypeId: setup.defaults.envTypes.ec2Linux.envTypeId,
+        envTypeConfigId: setup.defaults.envTypes.ec2Linux.envTypeConfigId,
+        studyIds: [studyId],
+        description: 'test',
+        projectId: setup.defaults.project.id,
+        cidr: '0.0.0.0/0',
+      });
+      // Poll until workspace is provisioned
+      await sleep(2000);
+      await adminSession.resources.workflows
+        .versions('wf-provision-environment-sc')
+        .version(1)
+        .findAndPollWorkflow(env.id, 10000, 60);
+
+      // Connect to workspace
+      const networkInfo = await admin2Session.resources.workspaceServiceCatalogs
+        .workspaceServiceCatalog(env.id) // env.id
+        .connections()
+        .connection('id-1')
+        .sendSshPublicKey({ keyPairId: keyPair.id });
+
+      await ssh.connect({
+        host: networkInfo.networkInterfaces[0].publicDnsName,
+        username: 'ec2-user',
+        privateKey: keyPair.privateKey,
+      });
+
+      // Mount studies
+      let output;
+      output = await mountStudies(ssh, studyId);
+
+      // Readwrite permission level
+      output = await readWrite(ssh, studyId);
+      expect(output.stdout).toEqual(expect.stringMatching(/ec2-user 20/));
```

Comment:
What does the `20` stand for?

Reply:
That's the file size created by the test using this [command](https://github.com/awslabs/service-workbench-on-aws/blob/f8311002b0c1be3f5a723e47951ba96ff3f764a0/main/integration-tests/support/complex/run-shell-command.js#L21). `ls -l` lists all files in the study folder with the size beside it, which is what is being checked here.

Reply:
Oh interesting. Without context the `20` isn't clear. Could we change the `readWrite` method to accept a parameter called `numberOfBytes`. Also lets let's a method description for that method to describe what it does in the `run-shell-command` file.

Then this will allow us to write this test code, which would be clearer.

```
const numberOfBytes = 20;
output = await readWrite(ssh, studyId, numberOfBytes);
expect(output.stdout).toEqual(expect.stringMatching(/`ec2-user ${numberOfBytes}`/));
```